q2_assembly/mason/mason.py

```diff
@@ -0,0 +1,289 @@
+        flags.extend([str(x) for x in arg_val])
+        return flags
+
+
+def generate_abundances(
+    profiles, num_genomes, mu=0, sigma=1, lambd=0.5, random_seed=42
+):
+    np.random.seed(random_seed)
+    all_abundances = []
+    for profile in profiles:
+        abundances = []
+        if profile == "uniform":
+            abundances = [1 / num_genomes] * num_genomes
+        elif profile == "exponential":
+            abundances = np.exp(-lambd * np.arange(num_genomes))
+            abundances /= abundances.sum()
+        elif profile == "lognormal":
+            abundances = np.exp(mu + sigma * np.random.randn(num_genomes))
+            abundances /= abundances.sum()
+        else:
+            print(f"Invalid abundance profile option: {profile}")
+        all_abundances.append(list(abundances))
+    return all_abundances
+
+
+def abundances_to_df(abundances, genome_files, sample_id):
+    data = {
+        "id": [os.path.basename(f).split(".")[0] for f in genome_files],
+        sample_id: abundances,
+    }
+    df = pd.DataFrame(data)
+    df.set_index("id", inplace=True)
+    return df
+
+
+def _combine_reads(sample_id, src_dir, orientation="forward"):
+    _idx = "1" if orientation == "forward" else "2"
+
+    all_files = sorted(
+        glob.glob(os.path.join(src_dir, f"{sample_id}_*_00_L001_R{_idx}_001.fastq.gz"))
+    )
+    with open(
+        os.path.join(src_dir, f"{sample_id}_00_L001_R{_idx}_001.fastq.gz"), "wb"
+    ) as out:
+        cmd = ["cat", *all_files]
+        subprocess.run(cmd, check=True, stdout=out)
+
+    # clean up
+    for f in all_files:
+        os.remove(f)
+
+
+def _process_sample(
+    sample, genome_files, abundances, total_reads, tmp_dir, threads, read_len, seed
+):
+    for genome_file, abundance in zip(genome_files, abundances):
+        genome_reads = int(total_reads * abundance)
+        _id = os.path.basename(genome_file).replace(".fasta", "")
+        print(f"[Sample: {sample}] Generating {genome_reads} reads for {_id}")
+
+        cmd = [
+            "mason_simulator",
+            "-v",
+            "--seed",
+            str(seed),
+            "--illumina-read-length",
+            str(read_len),
+            "--seq-technology",
+            "illumina",
+            "--num-threads",
+            str(threads),
+            "--input-reference",
+            genome_file,
+            "--num-fragments",
+            str(genome_reads),
+            "--out",
+            os.path.join(tmp_dir, f"{sample}_{_id}_00_L001_R1_001.fastq.gz"),
+            "--out-right",
+            os.path.join(tmp_dir, f"{sample}_{_id}_00_L001_R2_001.fastq.gz"),
+        ]
+
+        run_command(cmd, verbose=True)
+
+    # combine all reads into a single file and clean up
+    _combine_reads(sample, tmp_dir, orientation="forward")
+    _combine_reads(sample, tmp_dir, orientation="reverse")
+
+
+def _simulate_reads_mason(
+    reference_genomes: GenomeSequencesDirectoryFormat,
+    sample_names: List[str],
```

Comment:
```suggestion
    sample_names: List[str] = ["sample"],
```
I dont understand why you assign a default value for sample_names (sample_names = sample_names or ["sample"]) further down but not directly here.